addons/event/models/event_event.py

```diff
@@ -671,6 +671,13 @@ def _get_ics_file(self):
             result[event.id] = cal.serialize().encode('utf-8')
         return result
 
+    def _get_my_tickets_access_hash(self, registration_ids):
+        """ Returns the ground truth hash for accessing the tickets in route /event/my_tickets.
+        The dl links are always made event-dependant, hence the method linked to the record in self.
+        """
+        self.ensure_one()
+        return tools.hmac(self.env(su=True), 'event-registration-ticket-report-access', (self.id, sorted(registration_ids)))
```

Comment:
Instead of checking in controller registration_ids makes sense, maybe we could filter them here ? That way if trying with invalid registration_ids, you get no hash and it fails, something like that.

Reply:
Good idea. But I was wondering about using a link with several registrations including one that is deleted at some point. Would we not like to have the tickets of existing ones? I think it makes sense.

Since the link is created with all registration ids, the hash_truth should be computed using all registrations as well, so I still use them all, but only if at least one registration is valid. But I would still have to filter out non existing registrations in the controller to only consider actual registrations (and use attendee name in report name).

What do you think? We can also double up the checks but it would be twice the work ^^'

Reply:
Mmmmh you generally don't unlink registrations, but ok then.